functions-v2/src/on-analysis-document-pending.ts

```diff
@@ -84,30 +85,56 @@ export const onAnalysisDocumentPending =
       return;
     }
 
-    // Generate screenshot with Shutterbug service
+    // determine the summarizer to use, defaulting to "image"
+    let summarizer = queueDoc?.aiPrompt?.summarizer;
+    if (!summarizer && queueDoc?.firestoreDocumentPath) {
+      logger.info(`No summarizer specified, checking Firestore doc ${queueDoc.firestoreDocumentPath}`);
+      // until the unit configuration is updated to include the summarizer use the Firestore
+      // doc unit to see if we want to use the text summarizer (for the cas unit)
+      const firestoreDoc = await firestore.doc(queueDoc.firestoreDocumentPath).get();
+      const firestoreData = firestoreDoc.data();
+      if (firestoreData?.unit === "cas") {
+        logger.info(`Firestore doc ${queueDoc.firestoreDocumentPath} has unit "cas", using text summarizer`);
+        summarizer = "text";
+      }
+    }
+    summarizer = summarizer ?? "image";
+    let nextQueueDoc: Record<string, unknown> = {...queueDoc, summarizer};
 
-    let responseJSON;
-    try {
-      const html = generateHtml(JSON.parse(content));
-      const response = await fetch(shutterbugURL,
-        {
-          method: "POST",
-          body: JSON.stringify({content: html, height: 1500}),
-        }
-      );
-      responseJSON = await response.json();
-    } catch (err) {
-      await error(`Shutterbug error: ${err}`, event);
+    logger.info(`Using summarizer ${summarizer}`);
+
+    if (summarizer === "text") {
+      const docSummary = documentSummarizer(content, {});
+      nextQueueDoc = {...nextQueueDoc, docSummary};
+    } else if (summarizer === "image") {
+      // Generate screenshot with Shutterbug service
+      let responseJSON;
+      try {
+        const html = generateHtml(JSON.parse(content));
+        const response = await fetch(shutterbugURL,
+          {
+            method: "POST",
+            body: JSON.stringify({content: html, height: 1500}),
```

Comment:
I see the value existed previously, but am I understanding correctly that the 1500 is a generous estimate to ensure the entire content is _probably_ included?

Reply:
I'm guessing that is true.